src/provider/provider_os_memory_internal.h

```diff
@@ -67,10 +69,8 @@ typedef struct os_memory_provider_t {
     size_t partitions_weight_sum;
 
     hwloc_topology_t topo;
-    struct {
-        size_t allocated_memory;
-        size_t peak_memory;
-    } stats;
+
+    CTL_STATS;
```

Comment:
Why not: 
```cpp
ctl_stats_t stats;
```

Reply:
I agree with @KFilipek, reading through the code would be easier if `stats` wouldn't be hidden in a define